internal/scan-manager/scanners/qualys/src/main/java/org/wso2/security/tools/scanmanager/scanners/qualys/util/RequestBodyBuilder.java

```diff
@@ -0,0 +1,382 @@
+        type.appendChild(document.createTextNode(QualysScannerConstants.WAS_APP_REPORT));
+        report.appendChild(type);
+
+        Element config = document.createElement(QualysScannerConstants.CONFIG_KEYWORD);
+        report.appendChild(config);
+
+        Element webAppReport = document.createElement(QualysScannerConstants.WEB_APP_REPORT_KEYWORD);
+        config.appendChild(webAppReport);
+
+        Element target = document.createElement(QualysScannerConstants.TARGET);
+        webAppReport.appendChild(target);
+
+        Element webapps = document.createElement(QualysScannerConstants.WEBAPPS_KEYWORD);
+        target.appendChild(webapps);
+
+        Element webapp = document.createElement(QualysScannerConstants.QUALYS_WEBAPP_KEYWORD);
+        webapps.appendChild(webapp);
+
+        Element id = document.createElement(QualysScannerConstants.ID_KEYWORD);
+        id.appendChild(document.createTextNode(webAppId));
+        webapp.appendChild(id);
+
+        StringWriter stringWriter = buildSecureStringWriter(document);
+        createReportRequestBody = stringWriter.getBuffer().toString();
+
+        return createReportRequestBody;
+
+    }
+
+    /**
+     * Build launch scan request body.
+     *
+     * @param scanContext Qualys scanner parameters.
+     * @return Launch scan request body in XML format.
+     * @throws ParserConfigurationException Error occurred while parsing.
+     * @throws TransformerException         Error occurred while building secure string writer.
+     */
+    public static String buildScanLaunchRequest(ScanContext scanContext)
+            throws ParserConfigurationException, TransformerException {
+        String launchScanRequestBody;
+        DocumentBuilderFactory dbf = getSecuredDocumentBuilderFactory();
+        DocumentBuilder builder = dbf.newDocumentBuilder();
+        Document doc = builder.newDocument();
+        Element root = doc.createElement(QualysScannerConstants.SERVICE_REQUEST);
+        doc.appendChild(root);
+
+        Element data = doc.createElement(QualysScannerConstants.DATA);
+        root.appendChild(data);
+
+        Element wasScan = doc.createElement(QualysScannerConstants.WAS_SCAN);
+        data.appendChild(wasScan);
+
+        Element name = doc.createElement(QualysScannerConstants.NAME_KEYWORD);
+        name.appendChild(doc.createTextNode(
+                QualysScannerConstants.QUALYS_SCAN_NAME_PREFIX + scanContext.getWebAppName() + " " + getDate()));
+        wasScan.appendChild(name);
+
+        Element type = doc.createElement(QualysScannerConstants.TYPE_KEYWORD);
+        type.appendChild(doc.createTextNode(scanContext.getType()));
+        wasScan.appendChild(type);
+
+        Element target = doc.createElement(QualysScannerConstants.TARGET);
+        wasScan.appendChild(target);
+
+        Element webApp = doc.createElement("webApp");
+        target.appendChild(webApp);
+
+        Element webAppId = doc.createElement(QualysScannerConstants.ID_KEYWORD);
```

Comment:
Can't we move this element building part into a separate method and call it here by passing the tag name and the parent element since the same code repeats in several places.

Reply:
Request body pattern is different to each API invocation. One element is common to all request except the root element. It's better if we build complete request in each place for better readability.